spec.proto

```diff
@@ -0,0 +1,58 @@
+/**
+ * License goes here ....
+ */
+
+syntax = "proto3";
+
+package io.cloudevents;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_package = "io.cloudevents.proto";
+
+message CloudEvent {
+
+  // -- Standard CloudEvent Context Attributes
+
+  // Required
+  string id = 1;
+  string source = 2; // URI-reference
+  string spec_version = 3;
+  string type = 4;
+
+  // Optional
+  string data_content_type = 5; // RFC 2046
+  string data_schema = 6;       // URI
+  string subject = 7;
+  google.protobuf.Timestamp time = 8;
+
+  // -- CloudEvent Extension Attributes
+  map<string, CloudEventAttribute> extensions = 9;
```

Comment:
See also my comments in https://github.com/cloudevents/spec/issues/576#issuecomment-612107502

You may want to define all optional attributes (core and extension) as part of the same string-keyed map, rather than putting a few as tags but not later-defined attributes.

Reply:
that's an interesting point ... we don't do that in that in the other format definitions but is something to consider.

It doesn't have the potential to introduce complexity where consumers/produces might be using different version of the spec as new extensions are added - Proto can cope with that but it would increase complexity for consumers.

I'd argue for sticking with the defined 'required' and 'optional' context attributes.